MapboxCoreNavigation/RouteController.swift

```diff
@@ -859,51 +572,31 @@ extension RouteController: CLLocationManagerDelegate {
         }
     }
 
-    func getDirections(from location: CLLocation, completion: @escaping (_ route: Route?, _ error: Error?)->Void) {
+    func getDirections(from location: CLLocation, along progress: RouteProgress, completion: @escaping (_ route: Route?, _ error: Error?)->Void) {
```

Comment:
It's sort of strange that a routeprogress is needed to get directions. What if we rename this to

```swift
func getDirections(from location: CLLocation, along previousRoute: Route?, ...
```

Reply:
The problem is that the `RouteProgress.reroutingOptions(with:)` method relies on `RouteProgress.remainingWaypoints`.